Butler/#143

ButlerATTS opened the pull request:
The changes in this PR fix the bug described in this ticket: https://app.dework.xyz/alchemix/alchemix-v2-frontend/overview?taskId=af1e6eb3-8bac-488b-859a-82a0ac15f5ba

Specifically these changes:
(1) Refreshes user balances in when the users switch wallets
(2) Reloads the page when the user is switching networks within their wallet (for example changing from Ethereum to Optimism)

n4n0GH committed:
Merge pull request #443 from alchemix-finance/staging

Staging

n4n0GH committed:
Merge pull request #446 from alchemix-finance/staging

Staging

n4n0GH committed:
Merge pull request #448 from alchemix-finance/staging

Staging

n4n0GH committed:
Merge pull request #452 from alchemix-finance/staging

Staging

n4n0GH committed:
Merge pull request #454 from alchemix-finance/staging

Staging

n4n0GH committed:
Merge pull request #457 from alchemix-finance/staging

Staging

Commit:
Fix to #143 - Refreshing Balances When Switching Wallets

Added a call to fetchAllBalances() in the StateManager when the user switches addresses so that balances shown under vault strategies are synced to those of the wallet currently in use.

Commit:
Added a Page Reload When Switching Chains

Added a condition in the StateManager initialize function that will
reload the page if the user switches chains and the vaults were
previously loaded. This fixes an issue caused by the resetting of the
balances store where reactive components in the vaults page would query
properties of an empty balances store cuasing undefined errs that would
break the page.

n4n0GH committed:
Merge pull request #460 from alchemix-finance/staging

Staging

n4n0GH commented on src/components/composed/StateManager.svelte:
```diff
@@ -29,6 +29,11 @@
     if (!initStarted && $addressStore !== undefined && netId !== lastConnection.chainId) {
+      // Reload the page if the vaults were previously loaded and the user is switching chains
+      if (!$vaultsLoading && netId !== lastConnection.chainId) {
+        location.reload();
```
This is just doing a blatant refresh of the entire page which leads to a new batch or API requests being sent to the respective endpoints. We need a more elegant solution that doesn't create additional queries to keep within API request limits.

ButlerATTS replied:
I've removed this solution and fixed it by more thoroughly by removing the undefined reference errors that were causing this issue. Please see the latest commit for more details.

n4n0GH committed:
Merge pull request #463 from alchemix-finance/staging

Staging

n4n0GH committed:
Merge pull request #465 from alchemix-finance/staging

Staging

Commit:
Merge branch 'staging' into butler/#143

Commit:
Fixed undefined reference exceptions in reactive values on the Vaults page caused by resetting of data stores refreshing values after switching chains.

For object throwing these errs added undefined checks, and for numbers and BigNumbers
initialized values to 0 so that subsequent references to the value did
not result in additional errs that would break the page. These undefined
and default values are subsequently overwritten by an additional refresh
triggered by the fecthing of token balances which occurs right
afterwards.

ButlerATTS committed:
Merge branch 'alchemix-finance:prod' into butler/#143

Commit:
Cleaned up code by removing console.log() statements and a block of unecessary commented out code

n4n0GH commented on src/stores/v2/derived.ts:
```diff
@@ -76,15 +76,21 @@ export const vaultsAggregatedCoveredDebt = derived([vaultsStore, balancesStore],
-        return _prevVault.add(normalizedAmount);
-      }, initialVal),
+      [Number(vaultTypeKey)]:
+        (vaultBody)
```
I'm actually unsure if this check is necessary; line 68 already checks for `$vaults` so the const `{vaultBody}` should never be undefined.

ButlerATTS replied:
Could've sworn I saw this throwing an error at some point, but it looks like you're right. I reverted the code and ran tests on all button/toggle combinations for vault items and saw no issues. I've made a new push that reverts this to my branch, but I'm not sure if it's worth it to you for me to make a new PR for such a small change. Let me know
